src/components/SVGViewer/SVGViewerSearch.tsx

```diff
@@ -0,0 +1,351 @@
+      e.preventDefault();
+      this.props.openSearch();
+    }
+
+    if (e.keyCode === 13 && this.state.amountOfResults) {
+      this.handleNavigateSearchResult(1);
+    }
+  };
+
+  handleNavigateSearchResult = (move: number) => {
+    const { currentResult, amountOfResults } = this.state;
+    const { svg, zoomOnCurrentAsset } = this.props;
+    const searchResults = svg.querySelectorAll(`.${searchClassName}`);
+    const currentSearchResult = svg.querySelector(`.${currentSearchClassName}`);
+    if (!currentSearchResult) {
+      return;
+    }
+    let nextResult = currentResult + move;
+    if (move === 1 && nextResult > amountOfResults) {
+      nextResult = 1;
+    } else if (move === -1 && nextResult === 0) {
+      nextResult = amountOfResults;
+    }
+    currentSearchResult.classList.remove(currentSearchClassName);
+    searchResults[nextResult - 1].classList.add(currentSearchClassName);
+    this.setState({ currentResult: nextResult });
+    zoomOnCurrentAsset(searchResults[nextResult - 1]);
+  };
+
+  handleCancelSearch = () => {
+    this.resetSearch();
+    this.resetSearchResults();
+    this.props.handleCancelSearch();
+  };
+
+  handleSearchPhraseChange = (e: React.FormEvent<HTMLInputElement>) => {
+    const { value } = e.currentTarget;
+    const {
+      svg,
+      addCssClassesToMetadataContainer,
+      addCssClassesToSvgText,
+      zoomOnCurrentAsset,
+    } = this.props;
+    this.resetSearchResults();
+    if (!value) {
+      this.resetSearch();
+    } else {
+      const regexp = new RegExp(`(${value})`, 'gi');
+      let amountOfResults = 0;
+      let currentResult = 0;
+      const isMatchingSearchPhrase = (textNode: Element) =>
+        !!(textNode.textContent && textNode.textContent.match(regexp));
+      addCssClassesToMetadataContainer({
+        condition: isMatchingSearchPhrase,
+        className: searchClassName,
+      });
+      addCssClassesToSvgText({
+        condition: isMatchingSearchPhrase,
+        className: searchClassName,
+      });
+
+      // count all results and set current search result for navigation
+      const searchResults = svg.querySelectorAll(`.${searchClassName}`);
+      amountOfResults = searchResults.length;
+      if (amountOfResults) {
+        currentResult = 1;
+        searchResults.forEach(item => {
+          item.classList.remove(currentSearchClassName);
+        });
+        const currentSearchResult = searchResults[0];
+        currentSearchResult.classList.add(currentSearchClassName);
+        zoomOnCurrentAsset(searchResults[0]);
+      }
+      this.setState({
+        searchPhrase: value,
+        amountOfResults,
+        currentResult,
+      });
+    }
+  };
+}
+
+const StyledSearch = styled.div`
+  position: absolute;
+  top: 0;
+  right: 0;
+  z-index: 2;
+  display: flex;
+  align-items: center;
+  width: 100%;
+  height: 65px;
+  padding: 5px 5px 5px 15px;
+  border-radius: 0;
+  background: #ffffff;
+  border-bottom: 1px solid #e8e8e8;
```

Comment:
It's better to have the possibility to override these default colors.

Reply:
I made so a user can now provide a custom className for such things (see `CustomClassNames`).